src/main/java/de/clearit/kindergarten/appliance/vendor/VendorEditorView.java

```diff
@@ -43,23 +49,70 @@
   public VendorEditorView(VendorEditorModel model) {
     this.model = model;
   }
-
+  
   // Initialisation *********************************************************
   private void initComponents() {
     firstNameField = BasicComponentFactory.createTextField(model.getBufferedModel(VendorBean.PROPERTY_FIRST_NAME));
     lastNameField = BasicComponentFactory.createTextField(model.getBufferedModel(VendorBean.PROPERTY_LAST_NAME));
     phoneNumberField = BasicComponentFactory.createTextField(model.getBufferedModel(VendorBean.PROPERTY_PHONE_NUMBER));
-    vendorNumberField = BasicComponentFactory.createTextField(model.getVendorNumberFieldModel());
     vendorNumberTable = new StripedTable(new VendorNumberTableModel(model.getSelectionInList()));
     vendorNumberTable.setSelectionModel(new SingleListSelectionAdapter(model.getSelectionInList()
         .getSelectionIndexHolder()));
     addVendorNumberButton = new JButton(model.getAction(VendorEditorModel.ACTION_ADD_VENDOR_NUMBER));
     removeVendorNumberButton = new JButton(model.getAction(VendorEditorModel.ACTION_REMOVE_VENDOR_NUMBER));
+    vendorNumberField = BasicComponentFactory.createTextField(model.getVendorNumberFieldModel());
+  }  
+  
+  private void initFocusHandling() {
+	  vendorNumberField.addFocusListener(new FocusAdapter() {
+			@Override
+			public void focusLost(FocusEvent focusEvent) {
+				addVendorNumberButton.requestFocusInWindow();  
+			}
+	  });
   }
-
+  
+  private void initActionHandling() {
+	  ((JButton) addVendorNumberButton).addActionListener(new ActionListener() {
+		@Override
+		public void actionPerformed(ActionEvent e) {

```

Comment:
Was soll das hier machen?
Das soll doch dafür sorgen, dass nach dem Auslösen der "Verkäufernummer hinzufügen" Aktion, der Fokus wieder auf das Verkaeufernummerfeld gelegt werden soll, richtig?

Falls ja, dann kann das ganze eingedampft werden auf:

```java
addVendorNumberButton.addActionListener(new ActionListener() {

  @Override
  public void actionPerformed(ActionEvent e) {
    vendorNumberField.requestFocusInWindow();
  }

});  
```

P.S.: Wir muessen hier nicht explizit die Action in Model aufrufen, das passiert ja dadurch, dass der Button schon eine Action gesetzt bekommt in `initComponents()`.

Reply:
also was hier passiert ist das wenn man aus dem Eingabefeld -> Tab -> auf den Button das Enter ausführt, dieser entweder die Zahl (wenn eine Vorhanden) in die Tabelle setzt den Focus dann auf das Eingabefeld wieder setzt oder wenn das Feld "leer" ist das der Vendor gespeichert wird.

Wenn ich das einbaue hab ich den Focus mit Tab aus dem Eingabefeld nicht auf dem Button sondern direkt auf dem "OK" Button